waspc/waspls/test/Wasp/LSP/CompletionTest.hs

```diff
@@ -0,0 +1,157 @@
+module Wasp.LSP.CompletionTest where
+
+import Control.Lens ((^.))
+import Control.Monad.Log.Pure (runLog)
+import Control.Monad.State.Strict (evalStateT)
+import qualified Data.ByteString.Lazy as BS
+import qualified Data.ByteString.Lazy.Char8 as BSC
+import Data.Foldable (find)
+import Data.List (isPrefixOf)
+import Data.Maybe (mapMaybe)
+import qualified Language.LSP.Types as LSP
+import qualified Language.LSP.Types.Lens as LSP
+import System.FilePath (replaceExtension, takeBaseName)
+import Test.Tasty (TestTree, testGroup)
+import Test.Tasty.Golden (findByExtension, goldenVsStringDiff)
+import Text.Printf (printf)
+import Wasp.Analyzer.Parser.ConcreteParser (parseCST)
+import qualified Wasp.Analyzer.Parser.Lexer as Lexer
+import Wasp.LSP.Completion (getCompletionsAtPosition)
+import Wasp.LSP.ServerState (ServerState (ServerState, _cst, _currentWaspSource, _latestDiagnostics))
+
+-- | A string containing the input to a completion test. It represents wasp
+-- source code with a cursor position.
+--
+-- = Format
+--
+-- A normal wasp file, but with two addition
+-- 1) Begins with "//! test/completion" (this is future-proofing in case we
+--    add editor support for these test files).
+-- 2) In one spot in the file, add a "completion marker", which is a "|" with
+-- a "^" at the same column on the line below. The line containing "^" should be
+-- blank except for whitespace before the "^" and the "^" itself.
+--
+-- The "|", "^", and the extra line are not part of the wasp source code and
+-- are not included in the returned code. The preamble comment is included in
+-- the returned code.
+--
+-- If there is more than one completion marker in the input, only the first
+-- marker is recognized, and the rest are left in the source code.
+--
+-- === __Example__
+--
+-- @
+-- //! test/completion
+-- app todoApp {
+--   |
+--   ^
+-- }
+-- @
+--
+-- This test checks completions after the 2 spaces on the 2nd line.
+newtype CompletionTestInput = CompletionTestInput String
+
+-- | Run test cases in ./completionTests directory
+--
+-- See 'CompletionTestInput' for the format of the test input (stored in the
+-- .wasp files).
+test_CompletionLists :: IO TestTree
+test_CompletionLists = do
+  -- Path is relative to the root of the project, i.e. the directory containing
+  -- waspc.cabal.
+  completionTestInputFiles <- findByExtension [".wasp"] "./waspls/test/Wasp/LSP/completionTests"
+  return $
+    testGroup "Wasp.LSP.Completion.getCompletionsAtPosition" $
+      makeGoldenCompletionTest <$> completionTestInputFiles
+
+-- | Create a test case from a pair of <test_name>.wasp and <test_name>.golden
+-- file, using 'runCompletionTest' on the .wasp file to get the output to
+-- compare with or store in the .golden file.
+--
+-- The given filepath should be to a completion test input file (ending in .wasp)
+-- inside the completionTests folder.
+makeGoldenCompletionTest :: FilePath -> TestTree
+makeGoldenCompletionTest inputFp =
```

Comment:
```suggestion
makeGoldenCompletionTest testInputFile =
```

Two reasons for this change:
1. You use `Fp` suffix here, but not for `goldenFile`. So best to sync those. And Fp is  a bit harder to guess as to what it is.
2. testInput is more descriptive and we already use that term around.